### ISSUE TITLE
False positive reporting of SerializableClassMustDefineSerialVersionUID on interface

### ISSUE THREAD

**Issue by wmclifford** [CodeNarc/CodeNarc#639]
Not sure if this is working as intended or an oversight, but I was bitten by the `SerializableClassMustDefineSerialVersionUID` rule on a number of my interfaces which extend `Serializable`. Since they are interfaces, one would assume that they cannot define the `serialVersionUID` field, so the rule should, IMO, ignore this case:
```
interface X extends Serializable {
    // X's method definitions go here
}
```

Code in question from CodeNarc implementation:
```
class SerializableClassMustDefineSerialVersionUIDAstVisitor extends AbstractAstVisitor {
    @Override
    protected void visitClassEx(ClassNode node) {
        if (AstUtil.classNodeImplementsType(node, Serializable) && !node.isEnum()) {
            if (!node.fields.find { it.name == 'serialVersionUID' }) {
                addViolation(node, "The class $node.name implements Serializable but does not define a serialVersionUID")
            }
        }
        super.visitClassEx(node)
    }

}
```

I would imagine adding ` && !node.isInterface()` after the `!node.isEnum()` should fix the issue.

I can work around this for the moment by removing the `extends Serializable` from the interfaces in question and add `implements Serializable` to the classes which implement my interfaces, but I don't think this is the ideal solution. These interfaces/classes are to be used within a Jenkins pipeline, and so I am concerned that if I move the location of where things are declared to be `Serializable`, things will not work properly when the Jenkins job using the interfaces/classes executes.

Using tool version `2.1.0` with the Gradle `codenarc` plugin.

**Comment by chrismair** [comment]
Yup, that just looks like an oversight/bug.

**Comment by chrismair** [comment]
Thanks for the hint re: the fix. Will be available with CodeNarc 2.2.0.

**Comment by wmclifford** [comment]
Thanks @chrismair - looking forward to it.
